### ISSUE TITLE
Unable to save multicolor values if defaults didn't exist

### ISSUE THREAD

**Issue by smarterdigitalltd** [themeum/kirki#1813]
### Issue description:
Hi there. 

I am using some multicolor fields.

They work, but only if you apply default choices and only if you do this the FIRST time when adding the field.

If you do:

1) Add multicolor field with no default choices
2) Load page with Customizer
3) Modify multicolor field by adding default choices

Then you will still be unable to save values in Customizer. You cannot 'fix' the field by adding defaults later. 

Here is a GIF of what this looks like: https://cl.ly/0u142K03180b

It is not the most urgent issue, because now I know I must pass in some defaults (even if an empty string), but I just thought you should know :-)

Thanks for an awesome library...!

### Version used:
3.2.5 - via composer & separately plugin install

I have tried this on multiple brand new sites on different PHP versions, and it is the same behaviour.

### Using theme_mods or options?
theme_mods

### Code to reproduce the issue (config + field(s))

**Generic setup code:**

```
	$config = new stdClass();
	$config->customizer_config_id = 'bfm_customizer_config';
	$config->plugin_text_domain = 'bfm';
	$config->plugin_slug = 'bfm';

	\Kirki::add_config(
		$config->customizer_config_id, [
			'capability'  => 'edit_theme_options',
			'option_type' => 'theme_mod',
		]
	);

	if ( ! function_exists( 'wpd_add_customizer_field' ) ) {
		function wpd_add_customizer_field( $args ) {
			\Kirki::add_field( $config->customizer_config_id, $args );
		}
	}

	\Kirki::add_panel(
		$config->customizer_config_id . '_settings', [
			'priority'    => 10,
			'title'       => sprintf( __( '%s Settings.', $config->plugin_text_domain ), $config->plugin_name ),
		]
	);

	$settings_sections = [
		'menu_filter_typography'    => [ esc_attr__( 'Menu Filter Typography', $config->plugin_text_domain ), '' ],
		'menu_filter_buttons'       => [ esc_attr__( 'Menu Filter Buttons', $config->plugin_text_domain ), '' ],
	];

	foreach ( $settings_sections as $section_id => $section ) {
		$section_args = [
			'title'       => $section[ 0 ],
			'description' => $section[ 1 ],
			'panel'       => $config->customizer_config_id . '_settings',
		];

		if ( isset( $section[ 2 ] ) ) {
			$section_args[ 'type' ] = $section[ 2 ];
		}

		\Kirki::add_section( str_replace( '-', '_', $section_id ) . '_section', $section_args );
	}
```

**The field code (the code that causes the issue):**

```
	wpd_add_customizer_field(
		[
			'type'        => 'multicolor',
			'settings'    => 'menu_filter_button_border_color',
			'label'       => __( 'Button border color',  $config->plugin_text_domain ),
			'section'     => 'menu_filter_buttons_section',
			'choices'     => [
				'link'    => esc_attr__( 'Color', $config->plugin_text_domain ),
				'hover'   => esc_attr__( 'Hover', $config->plugin_text_domain ),
				'active'  => esc_attr__( 'Active', $config->plugin_text_domain ),
			],
		]
	);
```

**The updated field code (which works fine if you do this from the start, but not if you add 'default' array to an existing field later):**

```
	wpd_add_customizer_field(
		[
			'type'        => 'multicolor',
			'settings'    => 'menu_filter_button_border_color',
			'label'       => __( 'Button border color',  $config->plugin_text_domain ),
			'section'     => 'menu_filter_buttons_section',
			'choices'     => [
				'link'    => esc_attr__( 'Color', $config->plugin_text_domain ),
				'hover'   => esc_attr__( 'Hover', $config->plugin_text_domain ),
				'active'  => esc_attr__( 'Active', $config->plugin_text_domain ),
			],
			'alpha'       => true,
			'default'     => [
				'link'    => '',
				'hover'   => '',
				'active'  => '',
			],
		]
	);
```

**Comment by aristath** [comment]
All fields should have defaults. If no defaults have been defined then there will be issues.

If you have released version 1.0 of your theme without the defaults and you want to add them in v1.1 for example, then you might want to consider writing a migration function for that.
Shouldn't be too hard... something like this should work:

```php
add_action( 'wp', function() {
	$value = get_theme_mod( 'my_setting', array() );
	$initial_value = $value;
	if ( ! isset( $value['link'] ) ) {
		$value['link'] = '#000000';
	}
	if ( ! isset( $value['hover'] ) ) {
		$value['hover'] = '#333333';
	}
	if ( ! isset( $value['active'] ) ) {
		$value['hover'] = '#666666';
	}

	if ( $value !== $initial_value ) {
		set_theme_mod( 'my_setting', $value );
	}
} );
```
`wp` might not be the right place to do this, it depends on your scenario.

**Comment by smarterdigitalltd** [comment]
Thanks for your reply.
